### PR TITLE
Feature/group purposes

### PR DESCRIPTION
**Pull request by serhiihaniuk**

### CODE REVIEW

File: src/components/popup/details/purposes/purposes.jsx [hunk 1]
```diff
@@ -94,6 +96,42 @@ export default class Purposes extends Component {
 		return () => this.props.onShowVendors(filter);
 	}
 
+	isPublisherPurposeRequireConsent(purposeID) {
+		const { legIntPurposeIds, contractPurposeIds } = config;
+		if (!legIntPurposeIds || !contractPurposeIds) {
```
Comment (ghost) [inline src/components/popup/details/purposes/purposes.jsx:101]: it should works even we define `legIntPurposeIds` or `contractPurposeIds` only.

File: src/components/popup/details/purposes/purposes.jsx [hunk 2]
```diff
@@ -94,6 +96,42 @@ export default class Purposes extends Component {
...
+			return true;
+		}
+
+		//if purpose is in the list of legitimate interests or contract purposes then it does not require consent
+		return ![...legIntPurposeIds, ...contractPurposeIds].includes(purposeID);
+	}
+
+	groupPurposesByRequireConsent(allPurposes) {
+		const requireConsent = [];
+		const noConsent = [];
```
Comment (ghost) [inline src/components/popup/details/purposes/purposes.jsx]:
```suggestion
		const defaultConsent = [];
```

File: src/components/popup/details/purposes/purposes.test.js [hunk 3]
```diff
@@ -150,9 +150,9 @@ describe('Purposes and Features', () => {
 		purposes.handleSelectPublisherLegitimateInterest({isSelected: false, dataId: 2});
 
 
-		expect(selectPublisherLegitimateInterests.mock.calls[0][0]).to.equal(2);
+		expect(selectPublisherLegitimateInterests.mock.calls[0][0]).to.equal(1);
```
Comment (ghost) [inline src/components/popup/details/purposes/purposes.test.js:171]: Can you add test case for new "feature" ?

File: src/components/popup/details/purposes/purpose.jsx [hunk 4]
```diff
@@ -1,9 +1,8 @@
 import { h } from 'preact';
```
Comment (ghost) [inline src/components/popup/details/purposes/purpose.jsx:1]: I see additional letter below pubisher purposes list: 
image

File: src/components/popup/details/purposes/purposes.jsx (hunk 2 above)
Comment (serhiihaniuk) [reply src/components/popup/details/purposes/purposes.jsx]: - [ ]

File: src/components/popup/details/purposes/purposes.jsx (hunk 1 above)
Comment (serhiihaniuk) [reply src/components/popup/details/purposes/purposes.jsx:101]: ✅

File: src/components/popup/details/purposes/purposes.jsx (hunk 2 above)
Comment (serhiihaniuk) [reply src/components/popup/details/purposes/purposes.jsx]: ✅

File: src/components/popup/details/purposes/purposes.test.js (hunk 3 above)
Comment (serhiihaniuk) [reply src/components/popup/details/purposes/purposes.test.js:171]: ✅

File: src/components/popup/details/purposes/purpose.jsx (hunk 4 above)
Comment (serhiihaniuk) [reply src/components/popup/details/purposes/purpose.jsx:1]: ✅

File: src/components/popup/details/purposes/purposes.test.js [hunk 5]
```diff
@@ -161,27 +179,58 @@ describe('Purposes and Features', () => {
...
+				persistedConsentData={{}}
+				handleSelectTab={() => {}}
+				selectedTab={0}
+			/>,
+			scratch
+		);
+		expect(purposes.state.publisherPurposesToRender[0].id).to.equal(3);
+		expect(purposes.state.publisherPurposesToRender[2].id).to.equal(1);
+	});
 });
```
Comment (serhiihaniuk) [inline src/components/popup/details/purposes/purposes.test.js:236]: last one's new

**Review by ghost — APPROVED** [review]

**Pull request merged**